### PR TITLE
tool_operate: Fix --remote-time incorrect times on Windows. draft1

### PR DESCRIPTION
**Pull request by jay**
It was reported on the [mailing list](https://curl.haxx.se/mail/archive-2016-11/0033.html) that `--remote-time` isn't working properly on Windows and the timestamp may be off.

In the FAQ it [says](https://curl.haxx.se/docs/faq.html#Why_is_curl_R_on_Windows_one_ho):

> During daylight savings time, when -R is used, curl will set a time that appears one hour off.

--- 

curl is using utime to do the [modification](https://github.com/curl/curl/blob/curl-7_51_0/src/tool_operate.c#L1680-L1693). I tried these dates:

**Last-Modified: Thu, 01 Jan 1970 00:00:30 GMT**

curl filetime is 30 (correct), and it sets that using utime.
file's last modified FILETIME is 116444736300000000 (correct)

**Last-Modified: Thu, 20 Aug 1970 11:33:50 GMT**

curl filetime is 20000030 (correct) and it sets that using utime.
file's last modified FILETIME is 116644772300000000 (incorrect: Thu, 20 Aug 1970 12:33:50 GMT)
The FILETIME should actually be 116644736300000000.

---

Last-Modified is [defined](https://tools.ietf.org/html/rfc2616#section-14.29) HTTP-date and [must always be GMT/UTC](https://tools.ietf.org/html/rfc2616#page-21). So I think we could set the FILETIMEs directly on Windows just by using [SetFileTime](https://msdn.microsoft.com/en-us/library/windows/desktop/ms724933.aspx) instead.

My first take on this is attached, but it requires `i64` as a literal suffix and 64-bit integers.

---

Edit: Uploaded a build of draft1 at the reporter's request: 
[curl_pr1121_draft1.zip](https://github.com/curl/curl/files/590233/curl_pr1121_draft1.zip)

### CODE REVIEW

File: src/tool_operate.c [hunk 1]
```diff
@@ -1677,20 +1677,38 @@ static CURLcode operate_do(struct GlobalConfig *global,
...
+                                  116444736000000000i64;
+              HANDLE hfile = CreateFileA(outs.filename, FILE_WRITE_ATTRIBUTES,
+                                         (FILE_SHARE_READ | FILE_SHARE_WRITE |
+                                          FILE_SHARE_DELETE),
+                                         NULL, OPEN_EXISTING, 0, NULL);
+              if(hfile != INVALID_HANDLE_VALUE) {
+                SetFileTime(hfile, NULL, (FILETIME *)&converted,
+                            (FILETIME *)&converted);
+                CloseHandle(hfile);
+              }
```
Comment (bagder) [inline src/tool_operate.c]: Possibly showing an error message here in case of failure?

**Comment by bagder** [comment]
Oh, and maybe add a line to the FAQ about from which point in time this is fixed.

File: src/tool_operate.c (hunk 1 above)
Comment (jay) [inline src/tool_operate.c]: I notice we don't do that for utime so I didn't do it here, I'm not sure why. If it should be done for one should it be done for both? How about `fprintf(config->global->errors, "Failed to set modification times");`

Comment (bagder) [inline src/tool_operate.c]: Right, it should be done for both. I guess it happens so rarely people haven't noticed or cared.

Comment (jay) [inline src/tool_operate.c]: Ok. Also is there a preferred way to handle when 64-bit is available and what literal suffix to use? I'm not sure what is going to be portable across all Windows compilers. I guess I could use `curl_off_t` and `CURL_SUFFIX_CURL_OFF_T`

Comment (bagder) [inline src/tool_operate.c]: Yeah, those are the ones we have. Do we actually have non-64bit capable windows/compilers people still around?

Comment (jay) [inline src/tool_operate.c]: I don't know

**Comment by jay** [comment]
I am thinking I will modify my code so the fix is only present when there's a >= 64-bit curl_off_t, and otherwise fallback to utime. The alternative is add functions that can do 64-bit arithmetic using only 32-bit integers. Thoughts?

**Comment by bagder** [comment]
> I am thinking I will modify my code so the fix is only present when there's a >= 64-bit curl_off_t, and otherwise fallback to utime

I think that's fine. It should be relatively rare with <64 bit curl_off_t on windows builds.

**Pull request closed without merging**